cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -192,21 +222,36 @@ public void createTethering(FullHttpRequest request, HttpResponder responder, @P
   @POST
   @Path("/tethering/connections/{peer}")
   public void tetheringAction(FullHttpRequest request, HttpResponder responder, @PathParam("peer") String peer)
-    throws NotImplementedException, BadRequestException, PeerNotFoundException, IOException {
+    throws NotImplementedException, BadRequestException, IOException {
+    checkTetheringServerEnabled();
+
     String content = request.content().toString(StandardCharsets.UTF_8);
     TetheringActionRequest tetheringActionRequest = GSON.fromJson(content, TetheringActionRequest.class);
-    TetheringStatus tetheringStatus;
+    PeerInfo peerInfo;
+    try {
+      peerInfo = store.getPeer(peer);
+    } catch (PeerNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, GSON.toJson(new TetheringErrorResponse(e.getMessage())));
+      return;
+    }
+    if (peerInfo.getTetheringStatus() != TetheringStatus.PENDING) {
+      LOG.info("Action {} is not allowed in tethering state {}",
```

Comment:
this shouldn't be an info log. Any possible request should be treated as 'normal' behavior, as users can make any request that they want.

This seems more like one of the 40x response codes since it's not allowed to perform an action unless it's pending?

Reply:
returning BAD_REQUEST now.